src/loaders/OMETiffLoader.js

```diff
@@ -360,4 +352,46 @@ export default class OMETiffLoader {
       tiff.ifdRequests[index] = tiff.parseFileDirectoryAt(offsets[index]);
     }
   }
+
+  /**
+   * Trunactes tiles on the edge to match the height/width reported by the image.
+   * @param {tileData: TypedArray[]} data The array to be filled in.
+   * @param {Object} tile { x, y, z }
+   * @returns {TypedArray} TypedArray
+   */
+  _truncateTiles(data, tile) {
+    const { x, y, z } = tile;
+    const { tileSize } = this;
+    let height = tileSize;
+    let width = tileSize;
+    const {
+      height: zoomLevelHeight,
+      width: zoomLevelWidth
+    } = this.getRasterSize({ z });
+    const maxXTileCoord = Math.floor(zoomLevelWidth / tileSize);
+    const maxYTileCoord = Math.floor(zoomLevelHeight / tileSize);
+    if (x === maxXTileCoord) {
+      width = zoomLevelWidth % tileSize;
+    }
+    if (y === maxYTileCoord) {
+      height = zoomLevelHeight % tileSize;
+    }
+    if (
+      (width === tileSize && height === tileSize) ||
+      data[0].length === width * height
```

Comment:
why will we return early if `data[0].length === width * height`, regardless of `(width === tileSize && height === tileSize)`? It feels like we want to truncate always if width or height are not tilesize.